python_code_style.md

```diff
@@ -0,0 +1,119 @@
+# Python Code Style
+
+We use [pylint](https://www.pylint.org/) to perform automatic code checks on our
+repositories. It helps enforce code standard, and also does some static code
+analysis for potential programming errors.
+
+## Setting up a new project
+
+When creating a new repository that contains Python code, please add into the
+repository root folder a file named `.pylintrc` with the following contents:
+
+```
+[MESSAGES CONTROL]
+# We disable the following inspections:
+# 1. f-string-without-interpolation (we allow f-strings that don't do any 
+#    formatting for consistent looks and for future safety)
+# 2. inherit-non-class ("Inheriting 'NamedTuple', which is not a class" false 
+#    positive, see: https://github.com/PyCQA/pylint/issues/3876)
+# 3. too-few-public-methods (produces false positives)
+disable=f-string-without-interpolation,inherit-non-class,too-few-public-methods
+
+# Overriding these to allow short 1- or 2-letter variable names
+attr-rgx=[a-z_][a-z0-9_]{0,30}$
+argument-rgx=[a-z_][a-z0-9_]{0,30}$
+variable-rgx=[a-z_][a-z0-9_]{0,30}$
+
+# Maximum number of characters on a single line
+max-line-length=80
+
+# Set the linting for string quotes (for the plugin 
+# https://pypi.org/project/pylint-quotes/)
+string-quote=single
+triple-quote=double
+docstring-quote=double
+
+[MASTER]
+load-plugins=pylint_quotes
+```
+
+You may want to disable additional inspections if pylint hits a false positive.
+For example, it might fail to recognise imports of third-party libraries, in
+which case can add E0401 ("Unable to import") to the comma-separated list:
+
+```
+disable=f-string-without-interpolation,inherit-non-class,too-few-public-methods,E0401
+```
+
+In addition to `.pylintrc`, please create a GitHub Actions CI workflow under
+`.github/workflows/main.yaml` with the following contents (or just append the
+job called `lint`, if you have an exsting workflow):
+
+```
+name: CI
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+
+      - uses: actions/setup-python@v2
+
+      - name: Install pylint
+        run: pip install pylint pylint-quotes
+
+      - name: Run pylint
+        run: pylint **/*.py
+```
+
+This will make GitHub run Pylint on every push and pull request, and display
+checks in the web interface.
+
+<img src="assets/github_pylint_check.png" width="400"/>
+
+## Running Pylint
+
+To install Pylint into your environment, run:
+
+```
+pip install pylint pylint-quotes
```

Comment:
What does `pylint-quotes` do?

Reply:
Checks the consistency of quotes (single or double). I set up to single ones. Let me know if that's unnecessary!